lib/logic.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: false
+
+def bin_converter(state)
+  state.reduce(0) { |acc, value| acc * 2 + value }
+end
+
+WINNER_STATES = [
+
+  [1, 0, 0,
+   1, 0, 0,
+   1, 0, 0],
+
+  [0, 1, 0,
+   0, 1, 0,
+   0, 1, 0],
+
+  [0, 0, 1,
+   0, 0, 1,
+   0, 0, 1],
+
+  [1, 1, 1,
+   0, 0, 0,
+   0, 0, 0],
+
+  [0, 0, 0,
+   1, 1, 1,
+   0, 0, 0],
+
+  [0, 0, 0,
+   0, 0, 0,
+   1, 1, 1],
+
+  [1, 0, 0,
+   0, 1, 0,
+   0, 0, 1],
+
+  [0, 0, 1,
+   0, 1, 0,
+   1, 0, 0]
+
+].map { |state| bin_converter(state) }
+
+class GameState
+  def initialize
+    @board = [0, 0, 0,
+              0, 0, 0,
+              0, 0, 0]
+  end
+
+  # Will determine if it's a winning state
+
+  def winner?
+    b = bin_converter(@board)
+    WINNER_STATES.any? do |a|
```

Comment:
Here is the only place where you're using `WINNER_STATES`, I suggest you set as an attribute of **GameState** class.
Same thing for `bin_converter()`